### ISSUE TITLE
Pass original message in custom actions

### ISSUE THREAD

**Issue by shayne** [zabirauf/elixir_wit#4]
In your weather example, the `defaction fetch_weather/2` receives `session` and `context`. In your tests you're simply doing `Map.put(context, "forecast" "sunny")`, but how would I go about getting access to the entities to see if `location` was provided and if so, go hit an endpoint to pull the weather.

Wit.ai shows this example code:
```javascript
getForecast({context, entities}) {
  return new Promise(function(resolve, reject) {
    var location = firstEntityValue(entities, "location")
    if (location) {
      context.forecast = 'sunny in ' + location; // we should call a weather API here
      // ...
```

**Comment by zabirauf** [comment]
That is done in the `merge` step where the `merge` function of the action module is called. The function is passed the message from which you can extract the location from message and put it into context so you can later use it when the `fetch_weather` action is called.

You can check this example
https://github.com/zabirauf/witai_elixir_weather_bot/blob/master/lib/weather_conversation_action.ex

**Comment by shayne** [comment]
Looks like they removed the merge action by default: https://wit.ai/blog/2016/06/28/bot-engine-v1

> We removed the merge action to simplify stories. You can still include a merge action if you’d like, but this pattern is not forced anymore.

Merge wasn't running in my case, so I guess I have to figure out getting that action in my story. Would be nice for the `defaction` to get message as well, in the case of "What is the weather in Paris?", you would get the location without needing merge.

**Comment by zabirauf** [comment]
I wasn't aware of that. We already have the message available when calling the macro but just don't pass it in as it wasn't required before. Will make the change to pass the message in the action as well.

**Comment by zabirauf** [comment]
@shayne I've made the changes in commit f71a94d771e8de3714e2de1856ccb614fbcf8118 and also published the package v1.0.0 at https://hex.pm/packages/elixir_wit
You should be able to define custom actions as and use the message

```elixir
  defaction fetch_weather(session, context, message) do
    context # Return the updated context
  end
```
